distribution/push.go

```diff
@@ -107,7 +107,7 @@ func compress(in io.Reader) (io.ReadCloser, chan struct{}) {
 	pipeReader, pipeWriter := io.Pipe()
 	// Use a bufio.Writer to avoid excessive chunking in HTTP request.
 	bufWriter := bufio.NewWriterSize(pipeWriter, compressionBufSize)
-	compressor := gzip.NewWriter(bufWriter)
+	compressor, _ := gzip.NewWriterLevel(bufWriter, 1)
```

Comment:
This should be customizable?

Reply:
You may also want to take a look at containerd mode
https://github.com/containerd/containerd/blob/v2.0.0-rc.3/pkg/archive/compression/compression.go#L245